### PR TITLE
Add additional Cloud Datastore operators

### PR DESCRIPTION
**Pull request by turbaszek**
---
Make sure to mark the boxes below before creating PR: [x]

- [ ] Description above provides context of the change
- [ ] Unit tests coverage for changes (not needed for documentation changes)
- [ ] Target Github ISSUE in description if exists
- [ ] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [ ] Relevant documentation is updated including usage instructions.
- [ ] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by turbaszek** [comment]
@potiuk @mik-laj  should I split it to PR per operator?

**Comment by mik-laj** [comment]
@turbaszek  No. This is too much conflict because you are editing the same set of files.

File: airflow/providers/google/cloud/hooks/datastore.py [hunk 1]
```diff
@@ -104,14 +104,16 @@ def begin_transaction(self, project_id: str) -> str:
 
         :param project_id: Google Cloud Platform project ID against which to make the request.
         :type project_id: str
+        :param transaction_options: Options for a new transaction.
+        :type transaction_options: Dict[str, Any]
         :return: a transaction handle.
         :rtype: str
         """
         conn = self.get_conn()  # type: Any
```
Comment (michalslowikowski00) [inline airflow/providers/google/cloud/hooks/datastore.py:112]: Is it necessary to add such type notation like `type: Any`? Just asking. :)

Comment (turbaszek) [reply airflow/providers/google/cloud/hooks/datastore.py:112]: `Dict[str, Any]` is more informative than `Dict` because you know the type of keys

Comment (michalslowikowski00) [reply airflow/providers/google/cloud/hooks/datastore.py:112]: I was thinking about it `conn = self.get_conn()  # type: Any` :)

Comment (turbaszek) [reply airflow/providers/google/cloud/hooks/datastore.py:112]: Oh, I think it adds no value, but maybe it was required for mypy to be happy

File: airflow/providers/google/cloud/example_dags/example_datastore.py [hunk 2]
```diff
@@ -39,20 +42,121 @@
...
+    # [START how_to_rollback_transaction]
+    rollback_transaction = CloudDatastoreRollbackOperator(
+        task_id="rollback_transaction", transaction=begin_transaction_to_rollback.output
+    )
+    # [END how_to_rollback_transaction]
+
+
+if __name__ == "__main__":
+    dag2.clear(reset_dag_runs=True)
+    dag2.run()
```
Comment (turbaszek) [inline airflow/providers/google/cloud/example_dags/example_datastore.py:162]: Should be removed

**Pull request closed without merging**